### PR TITLE
Makefile: add gotag for make server_check

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What changed and how does it work?

There's some error check enabled by gotag intest.

And we're moving tests from Golang's unit test to the integration tests. We need to use the binary which is compiled with the gotag.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by winoros** [comment]
It's so surprising that some tests failed when we added the gotags.
I'll check it later.

**Review by AilinKid — APPROVED** [review]

**Review by okJiang — APPROVED** [review]
Maybe need a issue

File: pkg/executor/infoschema_reader.go [hunk 1]
```diff
@@ -308,7 +308,8 @@ func (e *memtableRetriever) setDataForVariablesInfo(ctx sessionctx.Context) erro
 
 func (e *memtableRetriever) setDataForUserAttributes(ctx context.Context, sctx sessionctx.Context) error {
 	exec := sctx.GetRestrictedSQLExecutor()
-	chunkRows, _, err := exec.ExecRestrictedSQL(ctx, nil, `SELECT user, host, JSON_UNQUOTE(JSON_EXTRACT(user_attributes, '$.metadata')) FROM mysql.user`)
+	wrappedCtx := kv.WithInternalSourceType(ctx, kv.InternalTxnOthers)
```
Comment (winoros) [inline pkg/executor/infoschema_reader.go:311]: It's a missed one. Found by the intest tag.

**Comment by winoros** [comment]
QQ_1721402706884
Now that the check_dev can pass.
We know that the failure is caused by
QQ_1721402744264

I'm trying to solve it without enlarging the cache size too much.

**Review by hawkingrei — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Review by D3Hunter — COMMENTED** [review]
can you separate the flags which is used for UT/FT and changes normal code path and the flag to control whether enable asserts which is harmless. i don't like the idea that we have to use 2 flags to do it for UT/FT

File: pkg/planner/core/operator/logicalop/logical_join.go [hunk 2]
```diff
@@ -1047,10 +1047,25 @@ func (p *LogicalJoin) ExtractUsedCols(parentUsedCols []*expression.Column) (left
...
+		if join.FullSchema != nil {
+			rSchema = join.FullSchema
+		}
+	}
 	for _, col := range parentUsedCols {
-		if lChild.Schema().Contains(col) {
+		if lSchema != nil && lSchema.Contains(col) {
 			leftCols = append(leftCols, col)
-		} else if rChild.Schema().Contains(col) {
+		} else if rSchema != nil && rSchema.Contains(col) {
```
Comment (winoros) [inline pkg/planner/core/operator/logicalop/logical_join.go:1068]: Found by integration test.

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**